lib/new_relic/traced_thread.rb

```diff
@@ -0,0 +1,38 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  #
+  # This class creates a thread that contains instrumentation to allow the agent to see spans created inside of this thread.
+  # In order to have this functionality inserted into all threads automatically,
+  # enable the `instrumentation.thread.tracing` configuration option in your newrelic.yml
+  #
+  # @api public
+  class TracedThread < Thread
+    #
+    # Creates a new thread that will be traced by the agent.
+    # Use this class the same as the Thread class
```

Comment:
```suggestion
    # Use this class as a replacement to creating Threads using `Thread.new`:
    # ```ruby
    #   TracedThread.new { execute_some_code }
    # ```
```

Reply:
Do you think users will be confused by the instrumentation being on `#initialize` rather than `#new`?